tests/components/bthome/test_binary_sensor.py

```diff
@@ -202,7 +202,7 @@ async def test_v2_binary_sensors(
     await hass.async_block_till_done()
 
 
-async def test_unavailable(hass: HomeAssistant) -> None:
+async def test_unavailable(hass: HomeAssistant, socket_enabled) -> None:
```

Comment:
Why does this need socket_enabled?

Reply:
Because I get errors on my system when running the tests if I leave this out. 

```
------------------------------------------------------------------------------------------------------------- Captured log call -------------------------------------------------------------------------------------------------------------
2023-12-15 23:14:31.655 INFO     SyncWorker_0 homeassistant.loader:loader.py:643 Loaded bthome from homeassistant.components.bthome
2023-12-15 23:14:31.663 INFO     SyncWorker_0 homeassistant.loader:loader.py:643 Loaded bluetooth_adapters from homeassistant.components.bluetooth_adapters
2023-12-15 23:14:31.675 INFO     MainThread homeassistant.setup:setup.py:301 Setting up bluetooth_adapters
2023-12-15 23:14:31.679 INFO     MainThread homeassistant.setup:setup.py:352 Setup of domain bluetooth_adapters took 0.0 seconds
2023-12-15 23:14:31.687 INFO     MainThread homeassistant.setup:setup.py:301 Setting up bthome
2023-12-15 23:14:31.688 INFO     MainThread homeassistant.setup:setup.py:352 Setup of domain bthome took 0.0 seconds
2023-12-15 23:14:31.709 INFO     SyncWorker_0 homeassistant.loader:loader.py:643 Loaded binary_sensor from homeassistant.components.binary_sensor
2023-12-15 23:14:31.715 INFO     SyncWorker_0 homeassistant.loader:loader.py:643 Loaded sensor from homeassistant.components.sensor
2023-12-15 23:14:31.721 INFO     MainThread homeassistant.setup:setup.py:301 Setting up binary_sensor
2023-12-15 23:14:31.723 INFO     MainThread homeassistant.setup:setup.py:352 Setup of domain binary_sensor took 0.0 seconds
2023-12-15 23:14:31.726 INFO     MainThread homeassistant.setup:setup.py:301 Setting up sensor
2023-12-15 23:14:31.729 INFO     MainThread homeassistant.setup:setup.py:352 Setup of domain sensor took 0.0 seconds
2023-12-15 23:14:31.776 INFO     MainThread homeassistant.components.binary_sensor:entity_platform.py:347 Setting up bthome.binary_sensor
2023-12-15 23:14:31.782 INFO     MainThread homeassistant.components.sensor:entity_platform.py:347 Setting up bthome.sensor
2023-12-15 23:14:31.815 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:714 Registered new sensor.bthome entity: sensor.test_device_18b2_pressure
2023-12-15 23:14:31.819 INFO     MainThread homeassistant.helpers.entity_registry:entity_registry.py:714 Registered new sensor.bthome entity: sensor.test_device_18b2_signal_strength
2023-12-15 23:14:31.865 INFO     MainThread habluetooth.scanner:events.py:80 hci0 (00:00:00:00:00:01): Bluetooth scanner has gone quiet for 90s, restarting
2023-12-15 23:14:31.889 WARNING  MainThread bluetooth_auto_recovery.recover:recover.py:461 Closing Bluetooth adapter hci0 [00:00:00:00:00:01] failed: 'NoneType' object has no attribute 'detach'
2023-12-15 23:14:31.899 ERROR    MainThread homeassistant:runner.py:137 Error doing job: Task exception was never retrieved:   File "/home/pi/core/venv/bin/pytest", line 8, in <module>
    sys.exit(console_main())
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 192, in console_main
    code = main()
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/config/__init__.py", line 169, in main
    ret: Union[ExitCode, int] = config.hook.pytest_cmdline_main(
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/main.py", line 318, in pytest_cmdline_main
    return wrap_session(config, _main)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/main.py", line 271, in wrap_session
    session.exitstatus = doit(config, session) or 0
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/main.py", line 325, in _main
    config.hook.pytest_runtestloop(session=session)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/main.py", line 350, in pytest_runtestloop
    item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 114, in pytest_runtest_protocol
    runtestprotocol(item, nextitem=nextitem)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 133, in runtestprotocol
    reports.append(call_and_report(item, "call", log))
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 222, in call_and_report
    call = call_runtest_hook(item, when, **kwds)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 261, in call_runtest_hook
    return CallInfo.from_call(
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 341, in from_call
    result: Optional[TResult] = func()
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 262, in <lambda>
    lambda: ihook(item=item, **kwds), when=when, reraise=reraise
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/runner.py", line 169, in pytest_runtest_call
    item.runtest()
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/python.py", line 1792, in runtest
    self.ihook.pytest_pyfunc_call(pyfuncitem=self)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_hooks.py", line 493, in __call__
    return self._hookexec(self.name, self._hookimpls, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_manager.py", line 115, in _hookexec
    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pluggy/_callers.py", line 77, in _multicall
    res = hook_impl.function(*args)
  File "/home/pi/core/venv/lib/python3.11/site-packages/_pytest/python.py", line 194, in pytest_pyfunc_call
    result = testfunction(**testargs)
  File "/home/pi/core/venv/lib/python3.11/site-packages/pytest_asyncio/plugin.py", line 525, in inner
    _loop.run_until_complete(task)
  File "/usr/lib/python3.11/asyncio/base_events.py", line 640, in run_until_complete
    self.run_forever()
  File "/usr/lib/python3.11/asyncio/base_events.py", line 607, in run_forever
    self._run_once()
  File "/usr/lib/python3.11/asyncio/base_events.py", line 1914, in _run_once
    handle._run()
  File "/usr/lib/python3.11/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/home/pi/core/tests/components/bthome/test_sensor.py", line 1207, in test_sleepy_device
    async_fire_time_changed(
  File "/home/pi/core/tests/common.py", line 428, in async_fire_time_changed
    _async_fire_time_changed(hass, utc_datetime, fire_all)
  File "/home/pi/core/tests/common.py", line 456, in _async_fire_time_changed
    task._run()
  File "/usr/lib/python3.11/asyncio/events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/lib/python3.11/asyncio/tasks.py", line 374, in create_task
    task = loop.create_task(coro)
Traceback (most recent call last):
  File "src/habluetooth/scanner.py", line 348, in _async_restart_scanner
  File "src/habluetooth/scanner.py", line 360, in habluetooth.scanner.HaScanner._async_restart_scanner
  File "src/habluetooth/scanner.py", line 376, in _async_reset_adapter
  File "/home/pi/core/venv/lib/python3.11/site-packages/habluetooth/util.py", line 13, in async_reset_adapter
    return await recover_adapter(adapter_id, mac_address)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/pi/core/venv/lib/python3.11/site-packages/bluetooth_auto_recovery/recover.py", line 377, in recover_adapter
    async with _get_adapter(hci, mac) as adapter:
  File "/usr/lib/python3.11/contextlib.py", line 204, in __aenter__
    return await anext(self.gen)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/pi/core/venv/lib/python3.11/site-packages/bluetooth_auto_recovery/recover.py", line 439, in _get_adapter
    await adapter.setup()
  File "/home/pi/core/venv/lib/python3.11/site-packages/bluetooth_auto_recovery/recover.py", line 185, in setup
    self.sock = btmgmt_socket.open()
                ^^^^^^^^^^^^^^^^^^^^
  File "/home/pi/core/venv/lib/python3.11/site-packages/btsocket/btmgmt_socket.py", line 68, in open
    sock_fd = socket.socket(AF_BLUETOOTH, SOCK_RAW, BTPROTO_HCI, fileno=fd)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/pi/core/venv/lib/python3.11/site-packages/pytest_socket.py", line 80, in __new__
    raise SocketBlockedError()
pytest_socket.SocketBlockedError: A test tried to use socket.socket.
```

Reply:
I see the same error here

https://github.com/home-assistant/core/actions/runs/7210822410/job/19645348027

Can it be that there is an issue introduced with https://github.com/home-assistant/core/pull/105685

Reply:
https://github.com/home-assistant/core/pull/105841

Reply:
I pulled the change in here and reverted the test changes to make sure it does fix the problem in the CI.

#105841 should merge first, and than we can pull the test change out of this PR

Reply:
Ok, I’ll wait till that one is merged. In both PR’s there is still one test failing, in the calendar integration.